ast.ts

```diff
@@ -76,7 +76,7 @@ export type Stmt<A> =
   | { a?: A; tag: "field-assign"; obj: Expr<A>; field: string; value: Expr<A> }
   | { a?: A; tag: "continue"; depth?: number }
   | { a?: A; tag: "break"; depth?: number } // depth is used for wasm 'br' instruction
-  | { a?: A; tag: "for"; name: string; index?: string; iterable: Expr<A>; body: Array<Stmt<A>> }
+  | { a?: A; tag: "for"; id: number; name: Destructure<A>; iterable: Expr<A>; body: Array<Stmt<A>> }
```

Comment:
@Quincy0v0 Is there a reason why we still need `id: number`, since any indices returned via `enumerate` would just be one of the `name: Destructure<A>`'s `targets`, right? Or am I misunderstanding something?

Reply:
This is not the same `id` here. This `id` is used to help us implement multiple for loops, since we are implementing for loops as objects and each for object requires an `id` to be properly saved.